src/main/java/games/strategy/util/Match.java

```diff
@@ -96,70 +89,45 @@ private Match(final Predicate<T> condition) {
 
   /**
    * returns true if all elements in the collection match.
+   * returns false if the collection is empty.
+   */
+  public static <T> boolean allMatchNotEmpty(final Collection<T> collection, final Match<T> match) {
```

Comment:
I agree, this method shouldn't be used.
But there are just too many calls to this method to check if the individual actually need the empty check. I'd suggest moving this change to another PR and making this method deprecated in the meantime. 
(I'm not going to rename it either, because I'm planning to remove it then.)